module/zfs/brt.c

```diff
@@ -1514,26 +1514,23 @@ brt_pending_remove(spa_t *spa, const blkptr_t *bp, dmu_tx_t *tx)
 void
 brt_pending_apply(spa_t *spa, uint64_t txg)
 {
-	brt_t *brt;
+	brt_t *brt = spa->spa_brt;
 	brt_pending_entry_t *bpe;
 	avl_tree_t *pending_tree;
-	kmutex_t *pending_lock;
 	void *c;
 
 	ASSERT3U(txg, !=, 0);
 
-	brt = spa->spa_brt;
+	/*
+	 * We are in syncing context, so no other brt_pending_tree accesses
+	 * are possible for the TXG. Don't need to acquire brt_pending_lock.
+	 */
```

Comment:
Maybe also ASSERT syncing context here, and !syncing in add/remove.

Reply:
I was thinking about it, but it is called so early that spa_syncing_txg is not even set yet, and I don't remember other ways to assert it.

Reply:
I'm not sure it's worth it, but you could move the call to `brt_pending_apply()` after `spa->spa_syncing_txg` is set in `spa_sync()`.  That downside is it would now happen under the SCL_CONFIG lock as a reader.